internal/controller/controllers/agent_controller_test.go

```diff
@@ -748,6 +748,66 @@ var _ = Describe("agent reconcile", func() {
 		Expect(agent.Status.DebugInfo.EventsURL).To(HavePrefix(expectedEventUrlPrefix))
 	})
 
+	It("validate Logs URL", func() {
+		serviceBaseURL := "http://acme.com"
+		hr.ServiceBaseURL = serviceBaseURL
+		hostId := strfmt.UUID(uuid.New().String())
+		expectedLogsUrlPrefix := fmt.Sprintf("%s/api/assisted-install/v2/clusters/%s/logs", serviceBaseURL, sId.String())
+		commonHost := &common.Host{
+			Host: models.Host{
+				ID:         &hostId,
+				ClusterID:  &sId,
+				Status:     swag.String(models.HostStatusKnown),
+				StatusInfo: swag.String("Some status info"),
+			},
+		}
+		backEndCluster = &common.Cluster{Cluster: models.Cluster{
+			ID: &sId,
+			Hosts: []*models.Host{
+				&commonHost.Host,
+			}}}
+		host := newAgent(hostId.String(), testNamespace, v1beta1.AgentSpec{ClusterDeploymentName: &v1beta1.ClusterReference{Name: "clusterDeployment", Namespace: testNamespace}})
+		By("before installation")
+
+		clusterDeployment := newClusterDeployment("clusterDeployment", testNamespace, getDefaultClusterDeploymentSpec("clusterDeployment-test", "test-cluster-aci", "pull-secret"))
+		Expect(c.Create(ctx, clusterDeployment)).To(BeNil())
+		mockInstallerInternal.EXPECT().GetHostByKubeKey(gomock.Any()).Return(commonHost, nil).AnyTimes()
+		mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(backEndCluster, nil).Times(1)
+		Expect(c.Create(ctx, host)).To(BeNil())
+		request := newHostRequest(host)
+		result, err := hr.Reconcile(ctx, request)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(result).To(Equal(ctrl.Result{}))
+		agent := &v1beta1.Agent{}
+		key := types.NamespacedName{
+			Namespace: testNamespace,
+			Name:      hostId.String(),
+		}
+
+		Expect(c.Get(ctx, key, agent)).To(BeNil())
+		Expect(agent.Status.DebugInfo.LogsURL).To(Equal(""))
+
+		By("during installation")
+		backEndCluster.Hosts[0].Status = swag.String(models.HostStatusInstalling)
+		backEndCluster.Hosts[0].LogsCollectedAt = strfmt.DateTime(time.Now())
+		mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(backEndCluster, nil)
+		_, err = hr.Reconcile(ctx, request)
+		fmt.Fprintf(GinkgoWriter, "agent logs url: %s\nhost id: %s\nevents url: %s", agent.Status.DebugInfo.LogsURL, hostId.String(), agent.Status.DebugInfo.EventsURL)
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(c.Get(ctx, key, agent)).ShouldNot(HaveOccurred())
+		Expect(agent.Status.DebugInfo.LogsURL).NotTo(BeEmpty())
```

Comment:
I don't think you need this expectation if you have the one on the following line.

Reply:
Gotcha, removed it. Thanks Nick!